app/api/reducers/dataReducer.js

```diff
@@ -55,6 +55,11 @@ function dataReducer(state = initialState, action) {
       return handleData(state, { resources });
     }
 
+    case actionTypes.RESERVATION_DELETE_SUCCESS: {
+      const reservation = Object.assign({}, action.payload, { state: 'cancelled' });
```

Comment:
Cancelling is done with a DELETE request and the API doesn't return anything with it. In the backend reservations are soft deleted so their state is just turned into 'cancelled'.
The payload here is the original reservation and put there by us in the deleteReservation action. I guess the state change could be made in the action as well if that would be more clear?
When fetching reservations or resources, the API also returns cancelled (deleted) reservations.